### ISSUE TITLE
Download binary to a non-system-wide directory that doesn't require sudo

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#10635]
Jesse Seldess (jseldess) commented:

Right now, our macOS and Linux binary install commands have the user copy the binary to a system-wide directory that often requires `sudo`. We should investigate another suitable location in the user's home direction.

cc @aaron-crl, @knz  

Jira Issue: DOC-1575

**Comment by emily-horing** [comment]
@jseldess have we made any progress on this? Wondering if we need to go with the sudo option or remove it with just a note to try it if it doesn't work for now.

**Comment by jseldess** [comment]
We haven't made any progress here. I think it can be a follow-up and we should go with `sudo` for now, as we do in the docs currently. 

cc @aaron-crl, @bdarnell for verification.

**Comment by bdarnell** [comment]
There are essentially three options: 
- We can use sudo to install to a system-wide directory on the PATH like `/usr/local/bin` and then refer to the binary with `cockroach sql`
- We can install without sudo to a directory under the user's home directory and include that directory whenever we refer to the binary (e.g. `~/bin/cockroach sql`). This leaks into all our other instructions. (but it's what I personally tend to do. I'd even translate our current instructions that recommend sudo and do this instead)
- We can install without sudo to a directory under the user's home directory and instruct the user to add that directory to their PATH so we can refer to it as `cockroach sql`. The problem with this is that there are a lot of ways to add things to PATH depending on the user's environment. 

My preference is for the binary to be installed system-wide, with a strong preference for installing with `brew` on macos, and eventually adding support for other package managers for other platforms.

**Comment by knz** [comment]
Ben I have noticed most Unix systems nowadays preconfigure PATH to include a standard subdirectory of $HOME. Can't we use that?

**Comment by bdarnell** [comment]
Yes, if there's a directory that is both writable and on PATH by default and is reasonably portable, we should definitely use that. But I'm not seeing it on the VMs I have lying around (mostly ubuntu 18.04). What is the directory you're seeing and on which distro?

**Comment by knz** [comment]
- on linux `~/bin` (traditionally) and `~/.local/bin`  (newer distros).

  The standard bashrc/zshrc etc only adds it if the directory exists, which might explain why you don't see it in your shell?

- on macos I don't think there's anything yet


Alternatively, we could do what other tech does and provide a shell script to install as e.g. `~/.cockroachrc` and tell folk to add `source ~/.cockroachrc` at the end of their shell initialization. We can even automate this setup by doing the `curl | bash` thing.

**Comment by jseldess** [comment]
I think this needs to have a product area home. @bdarnell, where do you think it lives? Is it most likely to be driven by CC given the primacy of the in-UI download experience there?
